AUTHORING_MANIFESTS.md

```diff
@@ -77,3 +82,42 @@ It is important to test your manifest before submission to ensure it meets the r
+
+### How to tell if it's a release candidate?
+
+- Most developers are very clear about pre-release and mainline versions. Ideally, this would be communicated to users through manifests and PackageIdentifiers. 
+- Some developers might add a common term such as "alpha", "beta", or "pre" - or an alphabet letter such as "b", or other indicator to their version numbers.
+- Other developers might add a different development line with a separate repo, different application names and icons, and separate version schema.
+- Not all letters indicate pre-release channels. For example, some developers append "-E" to the version number, to differentiate mainline Electron releases from their other mainline releases. If in doubt, ask the developer. 
+
+### Adding or removing letters from the version can cause matching issues
+
+- As long as a package version sticks to either semantic or string, it's pretty easy to sort. But switching usually causes the order to become confused.
+**This can cause the "upgrade always available" bug.**
+- This is another problem caused by adding pre-release versions into the mainline channel - it frequently disrupts the sort order, especially when the pre-release versions have letters in their version numbers, but the mainline versions do not.
+
+### How to make pre-release versions available
+
+Usually, just appending the appropriate term to the mainline PackageIdentifier is enough. The appropriate term would depend on how the developers refer to their early releases  - such as `.alpha`, `.beta`, `.pre`,  `.rc`, etc. And this new PackageIdentifier would replace the mainline PackageIdentifier in pre-release manifests, manifest names, and manifest paths. 
+
+Some examples: 
+
+- Mainline PackageIdentifier: `Microsoft.Edge`
+- Mainline Repository path: ./manifests/m/Microsoft/Edge
+- Beta PackageIdentifier: `Microsoft.Edge.Beta`
+- Beta Repository path: ./manifests/m/Microsoft/Edge/Beta
+
+- Mainline PackageIdentifier: `Microsoft.OpenSSH`
+- Mainline Repository path: ./manifests/m/Microsoft/OpenSSH
+- Beta PackageIdentifier: `Microsoft.OpenSSH.Beta`
+- Beta Repository path: ./manifests/m/Microsoft/OpenSSH/Beta
+
+Use the version number that the pre-release software adds to Apps and features Settings page, or other locations including the Registry, so that the WinGet package manager can provide updates to the pre-release version.
+
+Note: Pre-release PackageIdentifiers are subordinate to mainline PackageIdentifiers - if the pre-release PackageIdentifier is causing an issue in the mainline, then the mainline should have priority and the pre-release should be modified or removed. 
```

Comment:
It's important to note that we can only include the pre-release versions in the repository if there is some sort of metadata in the ARP that can distinguish the pre-release from the stable package. If both the variants use the same `DisplayName`, `ProductCode`, `Publisher` and other package matching related information, then an existing install of a stable package may be mapped to a higher available version of a pre-release version by WinGet, even if the PackageIds are different. A clear distinction in DisplayName can do & is more commonly.

This can be seen in the examples linked above

Microsoft.Edge:
	- DisplayName: `Microsoft Edge`
	  UpgradeCode: `'{883C2625-37F7-357F-A0F4-DFAF391B2B9C}'`

Microsoft.Edge.Beta:
	- DisplayName: `Microsoft Edge Beta`
	  UpgradeCode: `'{55B884A6-908D-3E59-BDCE-5E4BFA64FA7B}'`

This distinction would be enough for WinGet to see an installed package and determine exactly whether it should be mapped to `Microsoft.Edge` packageId or `Microsoft.Edge.Beta`

Reply:
I've been thinking over this, and am working on a very large rewrite.

Reply:
I forgot to add that currently we need some sort of ARP distinction is because we don't have native support for channels yet. We can add reference to https://github.com/microsoft/winget-cli/issues/147 in the rewrite for bulk of the work that's being tracked there